src/com/android/settings/wifi/WiFiTimeoutPreferenceController.java

```diff
@@ -0,0 +1,46 @@
+package com.android.settings.wifi;
+
+import android.content.Context;
+import android.provider.Settings;
+
+import androidx.preference.Preference;
+
+import com.android.settings.R;
+import com.android.settings.core.BasePreferenceController;
+
+public class WiFiTimeoutPreferenceController extends BasePreferenceController {
+
+    private final Context mContext;
+
+    public WiFiTimeoutPreferenceController(Context context, String key) {
+        super(context, key);
+        mContext = context;
+    }
+
+    public static long getWifiTimeoutDuration(Context context) {
+        return Settings.Global.getLong(context.getContentResolver(),
+                Settings.Global.WIFI_OFF_TIMEOUT, 0);
+    }
+
+    @Override
+    public int getAvailabilityStatus() {
+        return AVAILABLE;
+    }
+
+    @Override
```

Comment:
`super.updateState()` calls `refreshSummary()` already.
`refreshSummary()` calls `getSummary()` by default.

This code block should be replaced with
```
    @Override
    public CharSequence getSummary() {
        long currentTimeout = getWifiTimeoutDuration(mContext);

        return (currentTimeout != 0) ? 
            mContext.getString(R.string.wifi_timeout_summary, WifiTimeoutSettings.toHumanFriendly(currentTimeout)) : 
            mContext.getString(R.string.wifi_timeout_simple_summary);
    }
``

Reply:
done, thanks.